api/envoy/extensions/filters/listener/tls_inspector/v3/tls_inspector.proto

```diff
@@ -23,4 +23,11 @@ message TlsInspector {
 
   // Populate ``JA3`` fingerprint hash using data from the TLS Client Hello packet. Default is false.
   google.protobuf.BoolValue enable_ja3_fingerprinting = 1;
+
+  // The size in bytes of the initial buffer requested by the tls_inspector.
+  // If the filter needs to read additional bytes from the socket, the
+  // filter will double the buffer up to it's default maximum of 64KiB.
+  // If this size is not defined, defaults to maximum 64KiB that the
+  // tls inspector will consume.
+  google.protobuf.UInt32Value initial_read_buffer_size = 2 {lt: 65537 gt: 0}];
```

Comment:
Or alternatively, if the size is less than the (implementation-defined) size of a buffer slice (currently 4kb), just use the slice size as the initial value.

Reply:
The constraints should reflect what users should avoid from setting.
In this example, if the minimum number of bytes is at least 5, then that should be the lower-bound.
FWIW, in the future that lower-bound can be decreased (but not increased - as it may break valid configs).

Reply:
I bring up 5 bytes as the minimum it takes for the filter to determine it is not TLS, but that's only for certain data sequences such as all zero octets. In the expected case where the client is TLS, I'd think a minimum of 128 or 256 bytes would make sense. Anything lower than that won't provide any meaningful memory savings, but will require repeatedly getting more data and doubling the size to get up to something that could hold a meaningful portion of a ClientHello.